core/src/main/java/org/commonjava/indy/core/expire/ScheduleManager.java

```diff
@@ -699,34 +699,54 @@ private synchronized void removeCache( final ScheduleKey cacheKey )
     @CacheEntryCreated
     public void scheduled( final CacheEntryCreatedEvent<ScheduleKey, Map> e )
     {
-        final ScheduleKey expiredKey = e.getKey();
-        final Map expiredContent = e.getValue();
-        if ( expiredKey != null && expiredContent != null )
+        if ( e == null )
         {
-            logger.debug( "Expiration Created: {}", expiredKey );
-            final String type = (String) expiredContent.get( ScheduleManager.JOB_TYPE );
-            final String data = (String) expiredContent.get( ScheduleManager.PAYLOAD );
-            eventDispatcher.fire( new SchedulerScheduleEvent( type, data ) );
+            throw new IllegalArgumentException( "[FATAL]The infinispan cache created event for indy schedule manager is null." );
```

Comment:
Same thing here...we need to log to error and avoid throwing IllegalArgumentExceptions (at least, before we release this code).

I'm nervous about leaving these in because it would be easy to forget to change them...

Reply:
I'm not sure if this is a blocking error if this NPE happens, because if it happens all schedule works here will stop for indy. That's why I want to throw this IllegalArgumentException.
Maybe it's not a good way here to throw another Runtime error, I'll change it to an error log.